packages/eslint-plugin/src/configs/recommended.ts

```diff
@@ -0,0 +1,78 @@
+import { FlatConfig } from '@typescript-eslint/utils/ts-eslint';
+import globalIgnores from './ignores';
+import js from '@eslint/js';
+import globals from 'globals';
+
+/**
+ * Recommended ESLint configuration for JavaScript files.
+ */
+export default (plugin: FlatConfig.Plugin): FlatConfig.ConfigArray => [
+  globalIgnores,
+  js.configs.recommended,
+  {
+    name: '@tylertech-eslint/recommended-config',
+    languageOptions: {
+      globals: {
+        // Including globals from both browser and node environments for now, will reconsider later
+        // if we want to limit this to just browser or node or split into separate configs.
+        // We mostly work in TypeScript, and TypeScript will this for us.
+        ...globals.browser,
+        ...globals.node,
+      },
+    },
+    plugins: {
+      '@tylertech-eslint': plugin,
+    },
+    rules: {
+      // Recommended JavaScript rules
+
+      // Warn (all turned to error in strict config)
+      camelcase: 'warn',
+      'no-prototype-builtins': 'warn',
+      'no-case-declarations': 'warn',
+      'no-extra-boolean-cast': 'warn',
+      'no-async-promise-executor': 'warn',
+      'no-useless-escape': 'warn',
+      'no-loss-of-precision': 'warn',
+      'dot-notation': 'warn',
```

Comment:
I think the `tsRecommended` config should be disabling this and enabling `@typescript-eslint/dot-notation` instead.  Might want to audit the full set of [extension rules](https://typescript-eslint.io/rules?=extension#rules) to make sure no others were missed.

Reply:
@MikeMatusz Fixed in this commit, let me know if you find anything else!
https://github.com/tyler-technologies-oss/eslint-rules/pull/51/commits/349212d9a738f985524657d06b710ec0a34e6879